### PR TITLE
Issue 515  pytest 30

### PR DESCRIPTION
**Pull request by adler-j**
Updates to pytest 3.0 according to issue #515, also fixes issue #410

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -42,21 +59,25 @@ def pytest_addoption(parser):
...
 
 
-@pytest.fixture(scope="module", ids=ntuples_impl_ids, params=ntuples_impl_params)
+@pytest.fixture(scope="module", ids=ntuples_impl_ids,
+                params=ntuples_impl_params)
 def ntuples_impl(request):
+    """String with a available `NtuplesBase` implementation name."""
     return request.param
 
 ufunc_params = [ufunc for ufunc in odl.util.ufuncs.UFUNCS]
```
Comment (kohr-h) [inline conftest.py]: ??? `ufunc_params = odl.util.ufuncs.UFUNCS`?

File: odl/space/pspace.py [hunk 2]
```diff
@@ -154,20 +154,21 @@ class instance can be retrieved from the space by the
...
-
         See also
         --------
         ProductSpaceVectorWeighting
         ProductSpaceConstWeighting
 
         Examples
         --------
-        >>> import odl
+        Product of r2 and r3
```
Comment (kohr-h) [inline odl/space/pspace.py]: colon at the end

File: odl/space/pspace.py [hunk 3]
```diff
@@ -154,20 +154,21 @@ class instance can be retrieved from the space by the
...
+
         >>> r2x3 = ProductSpace(odl.rn(2), odl.rn(3))
 
+        Can also create power spaces
```
Comment (kohr-h) [inline odl/space/pspace.py]: Who can?
colon at the end

File: odl/space/pspace.py (hunk 2 above)
Comment (kohr-h) [inline odl/space/pspace.py]: Better: "Create the product of R^2 and R^3:"
Writing of the spaces is debatable, of course.

File: setup.py [hunk 4]
```diff
@@ -131,7 +131,7 @@ def run_tests(self):
     package_dir={'odl': 'odl'},
 
     install_requires=[requires],
-    tests_require=['pytest'],
+    tests_require=['pytest>3.0'],
```
Comment (kohr-h) [inline setup.py]: `>=`

**Comment by kohr-h** [comment]
Update also the `conda/meta.yaml` file. Speaking of which, I'd like to wait with the merge until version 3.0 of `pytest` hits the anaconda repos.

Apart from that, all cool.

File: conftest.py (hunk 1 above)
Comment (adler-j) [inline conftest.py]: *whistles*

**Comment by adler-j** [comment]
Short update: pytest still has not been updated to version 3.0 in conda.

I am considering if we should break this into two parts, one that gets rid of the 3.0 deprecation warnings, and one with the new functionality.

**Comment by kohr-h** [comment]
I know it's a pain having this PR lying around and assembling conflicts with master, but is getting rid of deprecation *warnings* really worth the trouble of splitting it? The pytest-3.0 specific part will have conflicts just in the same way. Honestly I think we can just wait.

**Comment by adler-j** [comment]
Conda has now been updated to the new version of pytest, I say we go ahead here. I'll fix this.

**Comment by kohr-h** [comment]
Ah, very good. I was already inclined to write that we go ahead anyway and recommend `pip install pytest` instead, but now that it's resolved, good.

**Comment by adler-j** [comment]
This merge would be unbearable, faster to redo this. Will fix.

**Comment by adler-j** [comment]
Closed in favor of new pull request

**Pull request closed without merging**